### ISSUE TITLE
White screen on start up

### ISSUE THREAD

**Issue by beall49** [MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-#908]
I received a white screen yesterday during playback. I closed and restarted, then reinstalled the app. I've reinstalled after uninstalling then downloading and cleaning out the old folders and still white screen on every launch. 

Windows 7 X64.

Here's the SquirrelSetup.log file, it appears to be having an issue with a .squirrel-lock file. I've tried to recreating that file, but it deletes it again on install.

[SquirrelSetup.txt](https://github.com/MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-/files/206826/SquirrelSetup.txt)

    2016-04-06 09:06:49> SingleGlobalInstance: Failed to grab lockfile, will retry: C:\Users\rbeall.NULAID\AppData\Local\Temp\.squirrel-lock-972EFCDE77036369B1DF56551A67239E566AF9BD: System.IO.IOException: The process cannot access the file 'C:\Users\rbeall.NULAID\AppData\Local\Temp\.squirrel-lock-972EFCDE77036369B1DF56551A67239E566AF9BD' because it is being used by another process.
       at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
       at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
       at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share)
       at Squirrel.SingleGlobalInstance..ctor(String key, TimeSpan timeOut)
    2016-04-06 09:06:49> SingleGlobalInstance: Failed to grab lockfile, will retry: C:\Users\rbeall.NULAID\AppData\Local\Temp\.squirrel-lock-972EFCDE77036369B1DF56551A67239E566AF9BD: System.IO.IOException: The process cannot access the file 'C:\Users\rbeall.NULAID\AppData\Local\Temp\.squirrel-lock-972EFCDE77036369B1DF56551A67239E566AF9BD' because it is being used by another process.
       at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
       at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
       at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share)
       at Squirrel.SingleGlobalInstance..ctor(String key, TimeSpan timeOut)
    2016-04-06 09:06:49> SingleGlobalInstance: Failed to grab lockfile, will retry: C:\Users\rbeall.NULAID\AppData\Local\Temp\.squirrel-lock-972EFCDE77036369B1DF56551A67239E566AF9BD: System.IO.IOException: The process cannot access the file 'C:\Users\rbeall.NULAID\AppData\Local\Temp\.squirrel-lock-972EFCDE77036369B1DF56551A67239E566AF9BD' because it is being used by another process.
       at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
       at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
       at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share)
       at Squirrel.SingleGlobalInstance..ctor(String key, TimeSpan timeOut)
    2016-04-06 09:06:49> SingleGlobalInstance: Failed to grab lockfile, will retry: C:\Users\rbeall.NULAID\AppData\Local\Temp\.squirrel-lock-972EFCDE77036369B1DF56551A67239E566AF9BD: System.IO.IOException: The process cannot access the file 'C:\Users\rbeall.NULAID\AppData\Local\Temp\.squirrel-lock-972EFCDE77036369B1DF56551A67239E566AF9BD' because it is being used by another process.
       at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
       at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
       at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share)
       at Squirrel.SingleGlobalInstance..ctor(String key, TimeSpan timeOut)
    2016-04-06 09:06:50> SingleGlobalInstance: Failed to grab lockfile, will retry: C:\Users\rbeall.NULAID\AppData\Local\Temp\.squirrel-lock-972EFCDE77036369B1DF56551A67239E566AF9BD: System.IO.IOException: The process cannot access the file 'C:\Users\rbeall.NULAID\AppData\Local\Temp\.squirrel-lock-972EFCDE77036369B1DF56551A67239E566AF9BD' because it is being used by another process.
       at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
       at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
       at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share)
       at Squirrel.SingleGlobalInstance..ctor(String key, TimeSpan timeOut)
    2016-04-06 09:06:50> SingleGlobalInstance: Failed to grab lockfile, will retry: C:\Users\rbeall.NULAID\AppData\Local\Temp\.squirrel-lock-972EFCDE77036369B1DF56551A67239E566AF9BD: System.IO.IOException: The process cannot access the file 'C:\Users\rbeall.NULAID\AppData\Local\Temp\.squirrel-lock-972EFCDE77036369B1DF56551A67239E566AF9BD' because it is being used by another process.
       at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
       at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
       at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share)
       at Squirrel.SingleGlobalInstance..ctor(String key, TimeSpan timeOut)
    2016-04-06 09:06:50> SingleGlobalInstance: Failed to grab lockfile, will retry: C:\Users\rbeall.NULAID\AppData\Local\Temp\.squirrel-lock-972EFCDE77036369B1DF56551A67239E566AF9BD: System.IO.IOException: The process cannot access the file 'C:\Users\rbeall.NULAID\AppData\Local\Temp\.squirrel-lock-972EFCDE77036369B1DF56551A67239E566AF9BD' because it is being used by another process.
       at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
       at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
       at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share)
       at Squirrel.SingleGlobalInstance..ctor(String key, TimeSpan timeOut)
    2016-04-06 09:06:50> SingleGlobalInstance: Failed to grab lockfile, will retry: C:\Users\rbeall.NULAID\AppData\Local\Temp\.squirrel-lock-972EFCDE77036369B1DF56551A67239E566AF9BD: System.IO.IOException: The process cannot access the file 'C:\Users\rbeall.NULAID\AppData\Local\Temp\.squirrel-lock-972EFCDE77036369B1DF56551A67239E566AF9BD' because it is being used by another process.
       at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
       at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
       at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share)
       at Squirrel.SingleGlobalInstance..ctor(String key, TimeSpan timeOut)
    2016-04-06 09:06:51> Unhandled exception: System.AggregateException: One or more errors occurred. ---> System.Exception: Couldn't acquire lock, is another instance running
       at Squirrel.SingleGlobalInstance..ctor(String key, TimeSpan timeOut)
       at Squirrel.UpdateManager.<acquireUpdateLock>b__1e()
       at System.Threading.Tasks.Task`1.InnerInvoke()
       at System.Threading.Tasks.Task.Execute()
    --- End of stack trace from previous location where exception was thrown ---
       at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
       at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
       at Squirrel.UpdateManager.<CheckForUpdate>d__0.MoveNext()
    --- End of stack trace from previous location where exception was thrown ---
       at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
       at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
       at Squirrel.Update.Program.<Download>d__5d.MoveNext()
       --- End of inner exception stack trace ---
       at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
       at System.Threading.Tasks.Task`1.GetResultCore(Boolean waitCompletionNotification)
       at System.Threading.Tasks.Task`1.get_Result()
       at Squirrel.Update.Program.executeCommandLine(String[] args)
       at Squirrel.Update.Program.main(String[] args)
    ---> (Inner Exception #0) System.Exception: Couldn't acquire lock, is another instance running
       at Squirrel.SingleGlobalInstance..ctor(String key, TimeSpan timeOut)
       at Squirrel.UpdateManager.<acquireUpdateLock>b__1e()
       at System.Threading.Tasks.Task`1.InnerInvoke()
       at System.Threading.Tasks.Task.Execute()
    --- End of stack trace from previous location where exception was thrown ---
       at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
       at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
       at Squirrel.UpdateManager.<CheckForUpdate>d__0.MoveNext()
    --- End of stack trace from previous location where exception was thrown ---
       at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
        at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
       at Squirrel.Update.Program.<Download>d__5d.MoveNext()<---

**Comment by jostrander** [comment]
Looks like you already have an instance of squirrel running, can you verify everything is closed out via the task manager? I'm not entirely familiar with Squirrel windows, but I think it shares with other electron apps on the system hence the error `SingleGlobalInstance`. 

@MarshallOfSound does that sound possible? If so apps like atom, slack, etc would need to be closed as well in order for squirrel to acquire a lock.

**Comment by MarshallOfSound** [comment]
Squirrel lock files should be independent of applications so other electron apps shouldn't be breaking this install.  Can you try deleting the lock file manually to try to reset it?

**Comment by beall49** [comment]
The lock file isn't even there....I tried creating it before install, and creating if after install with the player closed. In both instances the player deletes it (I'm guessing it deletes it)?

**Comment by jostrander** [comment]
Do you have hidden files turned on when you're looking for that?

**Comment by beall49** [comment]
Yeah, there's always on (I checked again just to make sure). I uploaded some screenshots so you guys can see what my screen looks like. I turned off slack to see if that had anything to do with, uninstalled and reinstall and still no luck. 

Question; if there was a network issue, do you think it could have anything to do with it? I haven't had anything else go wrong, but I don't control our web stuff here at work and I'm wondering if it's trying to do something that's making our web filter mad. IDK, just guessing now, this app has worked forever and now for it to just stop makes me think there's something else. 
![files](https://cloud.githubusercontent.com/assets/1039227/14332536/dc617a74-fbfe-11e5-9c14-f9f18fa1f30b.PNG)
![screen](https://cloud.githubusercontent.com/assets/1039227/14332539/e14669aa-fbfe-11e5-911a-c25920fa56f9.PNG)

**Comment by beall49** [comment]
I connected to my phone, reinstalled and it started right up. It must be something web related....not sure if you can but maybe throw an error on a future update letting us know our network sucks. Sorry I wasted your time :|

FYI, your app rocks.

Thanks again for your help, I appreciate it.

**Comment by jostrander** [comment]
Had a response half written out, but glad you got it! It's hard to say what it could have been, but the installer does connect and grab the nupkg from the releases page, as well as connect to @MarshallOfSound's website to query for the latest release, so could be one of those maybe.

**Comment by MarshallOfSound** [comment]
I tried running the application with a simulated speed of 10kb/s and with 0kb/s and couldn't replicate this issue.  Perhaps there was another underlying issue that fixed itself at around the same time you switched to your phone ? :laughing:

**Comment by beall49** [comment]
FYI, It's a new web filter setting at my work. 

I went directly to the play music site and it gave me ERR CONNECTION RESET errors. I emailed the sys admins and they found it being blocked and white listed the site(play music). I guess the filter blocked it because there's games on the play store, so the site falls under the gaming filter. Anyway, they removed the block and it works fine now. 

So not your fault, my bad.
